Remotion/Security/Core/ThreadPrincipalProvider.cs

```diff
@@ -33,11 +33,12 @@ public ThreadPrincipalProvider ()
 
     public ISecurityPrincipal GetPrincipal ()
     {
-      IIdentity identity = Thread.CurrentPrincipal.Identity;
+      // TODO: Add notnull assertions
+      IIdentity identity = Thread.CurrentPrincipal?.Identity!;
       if (!identity.IsAuthenticated)
         return s_nullSecurityPrincipal;
 
-      return new SecurityPrincipal (identity.Name, null, null, null);
+      return new SecurityPrincipal (identity.Name!, null, null, null);
```

Comment:
probably want to do an assert-not-null here since we don't control the IIdentity implementation. and we had an issue with someone implementing IIdentity the wrong way .